ui/app/src/modules/Content/Authoring/NewContentDialog.tsx

```diff
@@ -0,0 +1,447 @@
+    },
+    {
+      label: formatMessage(translations.contentTypeComponentLabel),
+      type: formatMessage(translations.contentTypeComponent)
+    },
+    {
+      label: formatMessage(translations.contentTypeQuickCreateLabel),
+      type: formatMessage(translations.contentTypeQuickCreate)
+    },
+    {
+      label: formatMessage(translations.contentTypeFavoriteLabel),
+      type: formatMessage(translations.contentTypeFavorite)
+    }
+  ];
+  const [error, setError] = useState<APIError>(null);
+  const resource = useStateResource<LegacyFormConfig[], EntityState<LegacyFormConfig>>(
+    filterContentTypes,
+    {
+      shouldResolve: (data) => !!data,
+      shouldReject: () => !!error,
+      shouldRenew: (source, resource) => resource.complete,
+      resultSelector: () => filterContentTypes,
+      errorSelector: () => error
+    }
+  );
+
+  const onTypeOpen = (srcData) => () => {
+    onDialogClose();
+    setDialogConfig({
+      open: true,
+      src: `${defaultFormSrc}?isNewContent=true&contentTypeId=${srcData.form}&path=${path}&type=form`
+    });
+  };
+
+  const onCompactCheck = () => setIsCompact(!isCompact);
+
+  const onResetFilter = useCallback(() => {
+    setResetFilterType(defaultFilterType);
+    setFilterContentTypes(contentTypes);
+  }, [contentTypes]);
+
+  const onTypeChange = useCallback(
+    (type) => {
+      resetFilterType && setResetFilterType('');
+
+      type !== defaultFilterType
+        ? setFilterContentTypes(contentTypes.filter((content) => content.type === type))
+        : onResetFilter();
+    },
+    [contentTypes, resetFilterType, onResetFilter]
+  );
+
+  const onSearch = useCallback(
+    (keyword) => {
+      const formatValue = keyword.toLowerCase();
+
+      !keyword
+        ? onResetFilter()
+        : setFilterContentTypes(
+            contentTypes.filter((content) => content.label.toLowerCase().includes(formatValue))
+          );
+    },
+    [contentTypes, onResetFilter]
+  );
+
+  const onSearch$ = useDebouncedInput(onSearch, 400);
+
+  const onSearchChange = (keyword) => {
+    setSearch(keyword);
+    onSearch$.next(keyword);
+  };
+
+  const onParentItemClick = (item) => {
+    setLoading(true);
+    setPreviewItem(item);
+  };
+
+  const getPrevImg = (content) =>
+    content?.imageThumbnail
+      ? `${contentTypesUrl}${content.form}/${content.imageThumbnail}`
+      : defaultPrevImgUrl;
+
+  const emptyStateSubtitle = () => (
+    <>
+      <FormattedMessage
+        id="newContentDialog.emptyStateMessageSubtitle"
+        defaultMessage="Try changing your query or browse the"
+      />{' '}
+      <Typography
+        variant="subtitle1"
+        component="a"
+        className={classes.emptyStateLink}
+        color="textSecondary"
+        onClick={onResetFilter}
+      >
+        <FormattedMessage
+          id="newContentDialog.emptyStateMessageLink"
+          defaultMessage="full catalog."
+        />
+      </Typography>
+    </>
```

Comment:
1. Move this inline on the render
2. Send the link as a `variable` to the formatted message component `variables`

```
<FormattedMessage
  id="newContentDialog.emptyStateMessageSubtitle"
  defaultMessage="Try changing your query or browse the <link>full catalog</link>."
  variables={{
    link: (msg) => 
      <Typography
          variant="subtitle1"
          component="a"
          className={classes.emptyStateLink}
          color="textSecondary"
          onClick={onResetFilter}
        >
          {msg}
        </Typography>
  }}
/>
```